doc/api/packages.md

```diff
@@ -10,87 +10,247 @@ A folder containing a `package.json` file, and all subfolders below that folder
 until the next folder containing another `package.json` file, are a _package
 scope_.
 
-## Determining module system
```

Comment:
I think “Determining module system” needs to come first, above any other section. The most important thing to know as a package author, even before the `package.json` fields, is that there are two module systems; and then when we introduce `"type"` and `"exports"` and those sections reference CommonJS and ESM, the reader already knows what those systems are.